### ISSUE TITLE
Recurring "Bluetooth keeps stopping" message on fresh install + other hardware issues

### ISSUE THREAD

**Issue by mrakow** [LineageOS/android_device_sony_pioneer#1]
After a fresh install of Lineage OS on the Sony Xperia XA2, there are severe hardware problems.

If I can help by providing more information, please advice what details you need.

## Buggy behaviour

Multiple hardware problems, most notably a "Bluetooth keeps stopping" alert pops up every 1-2 seconds. When tapping "Close app", its only button, the alert box reappears. This behaviour also occurs after enabling airplane mode.

There is other non-working hardware. I have not tried out all features, but below hardware is not functional:

*   Like Bluetooth, Wi-Fi does not turn on
*   The cameras don't work
*   headset is not recognised
*   Flash light cannot be turned on

## Steps to reproduce

1.   Install Lineage OS via TWRP
2.   Boot device

## Device and build info

Model: H3113
Bought from: [amazon.de](https://www.amazon.de/gp/product/B078KHZ5HG/)

Lineage OS build: [15.1-20180814](https://mirrorbits.lineageos.org/full/pioneer/20180814/lineage-15.1-20180814-nightly-pioneer-signed.zip)

TWRP version used: [twrp-3.2.3-0-pioneer-lineage.img](https://androidfilehost.com/?fid=3700668719832241508)

## Other potentially relevant info

*   The first installation of Lineage OS (using installation guide) soft-bricked the device and I had to flash a stock ROM using FlashTool. I haven’t thoroughly tested the functionality of the stock ROM, but no hardware issues were apparent. Starting from the re-flashed stock ROM, the second installation of Lineage OS seemed to be successful, except for the described hardware problems.

*   The exact same faulty behaviour occurred on the dual-SIM version. I had initially bought the dual-SIM version, but returned it, attributing the hardware issues to the non-matching installation image.

**Comment by luk1337** [comment]
do that:
```
dd if=/dev/block/bootdevice/by-name/dsp_a of=/dev/block/bootdevice/by-name/dsp_b
dd if=/dev/block/bootdevice/by-name/bluetooth_a of=/dev/block/bootdevice/by-name/bluetooth_b
dd if=/dev/block/bootdevice/by-name/modem_a of=/dev/block/bootdevice/by-name/modem_b
```

**Comment by mrakow** [comment]
I suspected that the drivers may be in the wrong A/B slot.

Do I enter the commands via adb shell?

**Comment by luk1337** [comment]
yes, after adb root.

**Comment by mrakow** [comment]
This seems to have worked (Bluetooth, wifi, camera etc. are OK after reboot). Thanks a lot.

**Comment by master-caster** [comment]
Hallo there,

how to gain root permission if the fresh installed system says **"su not found"**.
I granted root access for adb from the settings but can't execute the dd commands.

**Comment by luk1337** [comment]
adb wait-for-device root; adb wait-for-device shell.

**Comment by master-caster** [comment]
Thank you for your response but this doesn't work for me.
Do I need to install something like super su?

<pre>
benjamin@leopard:~/Downloads/platform-tools$ adb wait-for-device root
restarting adbd as root

benjamin@leopard:~/Downloads/platform-tools$ adb wait-for-device shell
H3113:/ $ su
/system/bin/sh: su: not found
127|H3113:/ $ dd if=/dev/block/bootdevice/by-name/dsp_a of=/dev/block/bootdevice/by-name/dsp_b
dd: /dev/block/bootdevice/by-name/dsp_a: Permission denied
1|H3113:/ $ ^D

benjamin@leopard:~/Downloads/platform-tools$ adb wait-for-device shell
H3113:/ $ dd if=/dev/block/bootdevice/by-name/dsp_a of=/dev/block/bootdevice/by-name/dsp_b
dd: /dev/block/bootdevice/by-name/dsp_a: Permission denied
1|H3113:/ $ 

</pre>

**Comment by luk1337** [comment]
oh right, latest build has broken adb root lmao, flash ` lineage-15.1-20181126-nightly-pioneer-signed.zip` instead.

**Comment by master-caster** [comment]
ah thank you will try that

**Comment by luk1337** [comment]
(((or just fastboot boot twrp and use adb shell there.)))

**Comment by master-caster** [comment]
okay than i'll try the last one first
xD

**Comment by master-caster** [comment]
so after it "stuck" in a boot loop for a while..... *raising heartbeat*

it started and it seems to work now
thank you @luk1337 for your fast guidance

**Comment by Th3M4ch1n3** [comment]
Hi guys, I'm having the same issue with sony_discovery. Problem is I cant event get ADB to see it anymore (presumably because I cant get past the setup to enable adb debugging). Can you guys help me out?

**Comment by luk1337** [comment]
use twrp + fix your drivers ^.^

**Comment by Th3M4ch1n3** [comment]
Couldnt even get to twrp as I couldnt get to bootloader (xa2 ultra...has a weird key combo) ended up just forcing my way through the setup and re-enabled adb.

**Comment by luk1337** [comment]
-- important info --

Whoever gets here, please follow https://forum.xda-developers.com/showpost.php?p=78973496&postcount=2 instead of copying bluetooth, dsp, modem partitions with dd.

-- important info --

**Comment by igoralmeida** [comment]
I have the same problem with H4113. Everything went fine until booting lineageos for the first time.

The original firmware was 50.1.A.13.83 if I'm not mistaken, and I never ran the original updater; just straight up unlocked, booted twrp and followed the instructions to sideload the 20190414 image.
I get the 'bluetooth keeps stopping' error, and wifi does not turn on.

I'm now trying with the 20190420 image to see if things change, but I'm not so sure. Could this be because the phone never had a chance to use both "boot slots"?

**Comment by luk1337** [comment]
...what about you follow the post above yours?

**Comment by luk1337** [comment]
Also protip: you want `50.1.A.13.123` instead of `50.2.A.0.*` when following that guide.

**Comment by igoralmeida** [comment]
I intend to, thank you. Maybe add that to the instructions to avoid tripping more people. If I hadn't taken notes of the original versions I could probably be going on a very different path to try to solve this.
Let's wait till it boots.

**Comment by igoralmeida** [comment]
Interesting, sideloading 20190420 seems to have fixed it. Bluetooth and wifi seem to be working fine.
I was surprised that the stock camera app did not show in the app list, but eventually I tested both that one and OpenCamera, and they're also working it seems.

Now that it's at least 'stable', I should try to update the firmware for `.123`.

**Comment by igoralmeida** [comment]
Thanks for the work by the way!

**Comment by luk1337** [comment]
No, it didn't fix it; you'll have same issue after next update if you don't follow #2.

**Comment by fdietze** [comment]
I had exactly this problem. After following the instructions here https://forum.xda-developers.com/showpost.php?p=78973496&postcount=2, and then installing lineage again, everything worked. Thank you @luk1337!

**Comment by throwaway3461984169** [comment]
I'm having the exact same problem with my new sony xperia xa2. I've spend hours to get the instructions working, but I always get an error with the newflasher:
No usb device with vid:0x0fce pid:0xb00b !
Can somebody help me please? Maybe @luk1337 have you seen this error before? Any help is appreciated!

**Comment by luk1337** [comment]
sudo

**Comment by throwaway3461984169** [comment]
Thank you for the suggestion, but I already ran it as root. Same error unfortunately.

**Comment by throwaway3461984169** [comment]
Is there maybe another way to solve the issue? It looks like there was another solution first in this thread.

**Comment by luk1337** [comment]
then your phone isn't in flash mode.

**Comment by throwaway3461984169** [comment]
I shutdown the phone, then pressed the volume down while plugging in the USB. The LED is green, so I think it should be in flash mode.
I tried on Linux and from a Windows VM if that makes any difference.

**Comment by luk1337** [comment]
dunno then

**Comment by throwaway3461984169** [comment]
If I understand correctly, this is a problem with missing drivers right? Can't those drivers be installed differently?
Otherwise, is only this particular model affected? Because then I might just send it back and buy a different one :/

**Comment by fdietze** [comment]
I did it on a Linux host inside a VirtualBox with Windows 10 installed. I remember that I also had Emma (https://developer.sony.com/develop/open-devices/get-started/flash-tool/download-flash-tool/) installed. Maybe that helps with the drivers. Does Emma detect your phone?

**Comment by throwaway3461984169** [comment]
Thank you for the suggestion @fdietze! I installed flashtool, ran the flashtool-drivers and afterwards the flashtool and it recognized my phone. Newflasher then recognized my phone too. I had to go to work though and aborted the newflasher and now it doesn't recognize the phone anymore. I tried to reproduce what I did this morning, flashtool still recognized my phone, but I can't get newflasher working again. I don't know what I'm doing wrong/differently. Do you remember the exact order in which you executed all steps?

**Comment by fdietze** [comment]
Following the steps was straightforward. At Step 4. I copied the Newflasher exe into the directory and double clicked it to execute. I executed all the fastboot commands from Linux.
Is your device detected when running `adb devices` or `fastboot devices`?

I also found that I could really switch the phone off when pressing both volume bottons and the power button at the same time for like 10 seconds. And then boot into your desired mode using the button combination.

I also set the usb-device in Virtualbox to permanent, so that after reconnecting the usb-cabel the phone is still connected to windows in the machine.

**Comment by throwaway3461984169** [comment]
The pressing of the volume buttons and power button simultaneously really did the trick. I was able to follow through with all instructions. Now my device won't boot though, it says the device is corrupt. Can I just sideload lineageos now or does this mean that the flashing of the firmware was unsuccessful?

**Comment by throwaway3461984169** [comment]
Well, it seems the flashing bricked the device :/ I followed exactly the tutorial, downloaded the latest firmware with XperiFirm (50.2.A.0.352 / R1B) and removed the files as listed in the tutorial, except two that were missing already:
- kernel_X-FLASH-ALL-18AE_0x00.hash
- kernel_X-FLASH-ALL-18AE.sin
I'm not sure what went wrong. Any advice on how I should proceed?

**Comment by fdietze** [comment]
The message about the device being corrupt can happen, but it does not necessarily mean that your device is bricked. I had such a message as well. I don't remember exactly what I did then...

Try to boot twrp with fastboot somehow and install lineageos.

**Comment by throwaway3461984169** [comment]
I thought I just lost 200€ haha. I was able to flash the stock rom with the sony flash tool, which btw didn't work out of the box and required a workaround. Now it's back to normal at least, except the unlocked bootloader. Fastbooting twrp only produces a black screen now (with light on).
I'm not sure yet how I will proceed. I might just leave the phone as it is, since every fix only created bigger problems so far. After 17 hours of struggle I'm kinda happy it works again.

Anyway, thanks so much for your help! This way I could at least try the fix and learned quite a bit about android.

Now, final question: Are only specific devices affected with the driver issue? Because just the sideloading of lineageos was actually pretty doable.

**Comment by fdietze** [comment]
I think it has to do with dual sim. A friend of mine got an XA2, but with single sim and didn't have these problems.

**Comment by throwaway3461984169** [comment]
I actually have the single sim model. Did your friend use the stock android for a while and made updates before sideloading lineage eventually? Or did he do it immediately after purchuase?

**Comment by fdietze** [comment]
He installed lineageos immediately after purchase.

**Comment by miezys** [comment]
I suspected that the drivers may be in the wrong A/B slot.

Do I enter the commands via adb shell?

@mrakow

I am having the same issue. Could somebody share commands used for copying drivers between slots? Thanks

**Comment by fdietze** [comment]
The instructions are here: https://forum.xda-developers.com/showpost.php?p=78973496&postcount=2

**Comment by miezys** [comment]
> 
> 
> The instructions are here: https://forum.xda-developers.com/showpost.php?p=78973496&postcount=2

Thanks for your reply. However, I strongly advise against following these instructions since I have bricked my XA2 (single sim) unrecoverable (the green LED blinks once when power button blinks, no fastboot, flashmode etc.).

**Comment by luk1337** [comment]
Something tells me that you didn't follow instructions the way they were written. eg. after first FW flash you booted directly to bootloader mode, switched slot and rebooted ^.^

**Comment by fdietze** [comment]
@miezys maybe that helps:
> I also found that I could really switch the phone off when pressing both volume buttons and the power button at the same time for like 10 seconds. And then boot into your desired mode using the button combination.

**Comment by hwesher** [comment]
Is the protip to install 50.1.A.13.123 instead of 50.2.A.0.* when following that guide still valid? XperiFirm does not list 50.1.A.13.123 to be available anymore - it only displays 50.2.A.0.* for download.

Regarding the installation instructions at xda-developers:  the directory where the firmware 50.2.A.0.* got downloaded does NOT contain the files "kernel_X-FLASH-ALL-18AE_0x00.hash" and "kernel_X-FLASH-ALL-18AE.sin". Any advice on how to deal with it?

**Comment by Matthias84** [comment]
I followed the instructions to downgrade the firmware with newflash/xperifirm to 50.1.A.13.123 to both slots and wiped and reinstalled LAOS 16 (20190708) including microg.

But the wifi is still offline. Can I somehow verify if the corresponding drivers etc.are present?

**Comment by fdietze** [comment]
If I remember correctly, I think did it without downgrading...

**Comment by luk1337** [comment]
[ boot stock; turn wifi on; reflash lineage; wifi works now; profit? ]

**Comment by Matthias84** [comment]
Today I flashed Android stock again using Sony flash tool Emma, rebooted the system, enabled wifi and reflashed LAOS. Now wifi etc. is working, thanks! :+1: 

_(The odd thing is, that I read about this issue before I got the phone. So I booted up the original android stock and connected wifi, did updates etc. But maybe, updating stock to Android 9.0 was raising this issue for my phone...)_

**Comment by Matthias84** [comment]
Now I noticed, that the screen rotate feature doesn't work. I tried to (re)enable it, but nothing happens. More over, if I install Sensor readout, it  doesn't list any sensors :cry: Is there an API change in Android9 or might this trouble be caused by this bug?

**Comment by luk1337** [comment]
https://gist.github.com/luk1337/d32876ec0fbcff4e91e1596554945284

**Comment by 234845** [comment]
Hello,

If use ADB sideload to install pioneer directly, it seems to be hit-or-miss if I get the 'bluetooth keeps stopping' problem. If I force select slot a before booting into TWRP (using the fastboot command) I *always* end up with the 'Bluetooth keeps stopping' problem, but if I force select slot b I never get the problem (at least in the last 8 re-installs I have just done to test this theory).

So, for anyone who is interested (take it or leave it), I now have a procedure that takes me less than 10 mins and (for me anyway) works every time, requires no complicated flashing of firmware.

Here are the details of what I am using and the few steps to achieve it (note I always wipe clean my entire device - it's easier, I have few apps to reinstall). I also use Linux.

- Linux
- Android Debug Bridge version 1.0.40 - Version 28.0.2-5303910
- lineage-16.0-20190720-nightly-pioneer-signed.zip
- twrp-3.2.3-1-pioneer.img

Steps

On the PC, with the unlocked phone switched on and connected via USB, open a terminal and type:<br>
`adb device`

> `List of devices attached`<br>
> `CQ30017Q8P	device`

Fastboot the phone into TWRP by running command:<br>
`adb reboot bootloader`

Wait for blue light on phone (top right-hand corner)

Verify fastboot has worked by running command:<br>
`fastboot devices`

> `CQ30017Q8P	fastboot`

Set active slot to 'b' by running command:<br>
`fastboot set-active=b`

>  `Setting current slot to 'b'  OKAY [  0.006s]`<br>
>  `Finished. Total time: 0.006s`


Boot into twrp by running command:<br>
`fastboot boot 'path_to_twrp-x.x.x-x-pioneer.img'`

>`Sending 'boot.img' (57720 KB)   OKAY [ 1.518s]`<br>
>`Booting              OKAY [ 5.130s]`<br>
>`Finished. Total time: 6.699s`<br>


Phone will now reboot into TWRP

Swipe to a allow modifications

Tap 'Wipe'.

Select all boxes except USB-OTG then 'Swipe to Wipe' (will only take a few seconds)

Tap Android 'Back' button (3 times) to take you back to main page

Tap 'Advanced'

Tab 'ADB sideload'

Check 'Wipe Dalvik Cache' and 'Wipe Cache' boxes

Now 'Swipe to Start Sideload'.

On PC begin sideloading the package by running the command:<br>
`adb sideload path_to_lineageos_download.zip`

This will take approximately 5 minutes.

> `adb sideload lineage-16.0-20190323-nightly-pioneer-signed.zip`<br> 
> `serving: 'lineage-16.0-20190323-nightly-pioneer-signed.zip' (~7%)`<br>
> `serving: 'lineage-16.0-20190323-nightly-pioneer-signed.zip' (~8%)`<br>
> `serving: 'lineage-16.0-20190323-nightly-pioneer-signed.zip' (~8%)`<br>
> `...`<br>
> `serving: 'lineage-16.0-20190323-nightly-pioneer-signed.zip' (~99%)`<br>
> `...`<br>
> `Total xfer: 1.00x`<br>

Tap Reboot System

> **Important Note:**<br>
> The phone will warn about '**No Operating System!**'. Ignore this and swipe to reboot phone anyway.

Wait for phone to reboot and walk through the normal LineageOS startup configuration (Language, time/date, Wifi etc.)

**Comment by joewa** [comment]
Yes I am very interested! I am happy we almost have it!

But **one issue still remains**:
**Every OTA update is going to break it again**

Here are my findings:
- I noticed (before reading your manual @234845 ) that `adb sideload` seems to work every 2nd time. So the issue is apparently related to the the slot A/B switching.
- I had a working LOS `lineage-16.0-20190719-nightly-pioneer-signed.zip` yesterday.
- OTA upgrade is a great feature which I wanted to test: I did an upgrade to get today's nightly. This breaks everything again.
- After wiping and sideloading `lineage-16.0-20190720-nightly-pioneer-signed.zip` everything works again.

So, **don't touch the (OTA) upgrade button**...

**Any idea how to fix the OTA upgrades?**

**Comment by 234845** [comment]
I haven't tested this but perhaps after updating you can fastboot, set slot B again, reboot as normal and then see if OTA works? This is pure speculation but why not give it a try?

I know it's going to be a pain (if it does work) but at least that way you might be able to OTA.

**Comment by joewa** [comment]
Hi @234845 and All,
I suppose this cannot work because the OTA updater will always copy the update to the other slot and switch to this slot when rebooting. Changing the slot afterwards will likely break it (haven't tested).

Having this behaviour in mind **I found a solution which works for me** now:
- Wait until at least two updates are available (which is 2 days for the nightly)
- Download both using the OTA updater
- First install the older one and reboot. Be brave because your Bluetooth and Wifi is broken again.
- Navigate to the updater (while confirming the repeatedly upcoming complains about Bluetooth)
- Now install the other most recent update and reboot again.

Your phone should be working with a fresh Lineageos now :smiley_cat: 

Of course it would be better to fix the root cause of this mess. I didn't go through the procedure here: https://forum.xda-developers.com/showpost.php?p=78973496&postcount=2 because
- I am not sure if it fixes the problem with OTA update later
- People seem to be not successful using it recently

**Comment by fdietze** [comment]
> * I am not sure if it fixes the problem with OTA update later
It solves it for me. I have no problem installing updates.

**Comment by vog** [comment]
While I'm glad that there is some solution to this problem, the situation is still very unsatisfying, especially with regard to the LineageOS goal of supporting devices for long time periods:

1. Either that forum entry, or this issue entry, should be linked from the installation instructions (https://wiki.lineageos.org/devices/pioneer), because literally everybody trying to install LineageOS on Sony Xperia XA2 will be haunted by this problem.

2. The solution depends on the [XperiFirm](https://forum.xda-developers.com/showpost.php?p=78973496&postcount=2) tool. While it is great that this tool exists at all, it is a proprietary tool maintained by a single person who didn't provide any sources. What if Sony changes their firmware download structure? Will that person always find the time to quickly adjust their tool, even in 5-10 years from now?

3. The solution depends on the availability of an old 50.1.xxx firmware image by Sony. How long will they provide it? For example, for my model there are 18 "Operator" variants available, 15 of which are already on 50.2.xxx. What if the remaining 3 are updated, too?

4. There is no legal way to provide a mirror those firmware images, which seems to be the only long-term fix for issues 2+3.

4. The solution depends on the [newflasher](https://github.com/munjeni/newflasher) tool. While that tool is great and Open Source, it currently needs to be compiled by hand and it is unclear if that will ever be packaged by the distros.

In short, can't we do better here? Is there, perhaps, any hope to solve this issue directly within LineageOS?

**Comment by slashr** [comment]
@234845 solution worked for me!

**Comment by vog** [comment]
Okay, so this is my take on the issues identified in my [previous comment](https://github.com/LineageOS/android_device_sony_pioneer/issues/1#issuecomment-517958442).  This is an overview, more details will follow:

1. *Documentation:* I created detailed step-by-step instructions, all in one place, that take anyone from the stock firmware phone to a working LineageOS with flawless OTA updates. I even automated ~95% of those in a script and verified it on various Xperia XA phones. Along the way, I noticed various shortcomings and errors in existing instructions that made those work sometimes but not always.

2. *Dependency on proprietary XperiFirm tool:* I created a simple shell script that downloads, merges and unpacks the firmware images as needed. It works for every Xperia XA2 model (H3113, H3123, H3133, H4113, H4133) and is thus a 100% replacement for XperiFirm, at least for our purposes. In addition, it reduces the unpacked firmware to the subset needed to fix the Bluetooth/Wifi bug for you (as described in https://forum.xda-developers.com/showpost.php?p=78973496&postcount=2), so its output can be used directly in [newflasher](https://github.com/newflasher/newflasher).

3. *Availability of 50.1.xxx firmware versions:* I noted that for each model, the firmware for "Commercial and Journalists" appears to be frozen on version 50.1.A.5.59, which is not the newest 50.1.xxx firmware, but recent enough to fix the Bluetooth/Wifi bug. So I hardcoded those in the above mentioned script. This has the additional advantage that every future user of this script will fetch the exact same version that I thoroughly tested.

4. *Availability of firmware downloads:* I don't have a final solution to this, but the above mentioned script verifies all downloaded firmware pieces by their SHA-256 hashes, which are hard coded into the script. So in the worst case anyone could create a mirror anywhere, and future users could still download the firmware safely.

5. *Maintenance of newflasher tool:* Until someone else wants to take over, I'm [volunteering to maintain the Debian package of newflasher](https://github.com/newflasher/newflasher/issues/9) and coordinated everything with the respective Debian team. Along the way, I also became the second upstream maintainer of newflasher, in tight cooperation with the newflasher owner.

Open question:

* How do I integrate my documentation with the official LineageOS instructions? To be 100% accurate, I'd have to change existing (templated) instructions as well, but that would affect all models which is not desirable.  On the other hand, replacing the official instructions completely with my own (or a link to it) would be considered a bit rude, wouldn't it?

**Any support, suggestions, ideas and advice are welcome!**

**Comment by throwaway3461984169** [comment]
@vog That sounds awesome, thank you for work! I'd love to give this a try. Can you provide a link to your script and instructions in this thread?

**Comment by vog** [comment]
Thanks for your interest. I'm about to polish up everything for publication, and will post the link here as soon as I published it.

**Comment by malakai97** [comment]
@vog It'd be great if you could publish whatever you have.

**Comment by vog** [comment]
@malakai97 Thanks for your interest! I'll publish what I have as soon as possible.

**Comment by vog** [comment]
Okay, I just published the tool. Sorry for the long delay!

**Ait – Android installation tool**

Project website with instructions:
* https://v0g.gitlab.io/ait/

GitLab project website:
* https://gitlab.com/v0g/ait

@throwaway3461984169, @malakai97: It would be great to get some feedback from you!

**Comment by luk1337** [comment]
/me yawns

BTW 50.2 is now supported b-but if you're on 50.1 you'll have to flash magic zip in twrp, see https://forum.xda-developers.com/showpost.php?p=78973496&postcount=2

**Comment by munzli** [comment]
> Okay, I just published the tool. Sorry for the long delay!
> 
> **Ait – Android installation tool**
> 
> Project website with instructions:
> 
> * https://v0g.gitlab.io/ait/
> 
> GitLab project website:
> 
> * https://gitlab.com/v0g/ait
> 
> @throwaway3461984169, @malakai97: It would be great to get some feedback from you!

@vog Thanks to your `ait` script i was able to find what was needed to get my XA2 to work flawlessly!
with the `download-firmware-reduced` command and using `newflasher` on both slots with the firmware files in the firmware-H3113-reduced folder now both slots boot and work great!
Thank you so much :+1:

**Comment by Ygarr** [comment]
Hello,234845
And how to install gapps in what slot?

**Comment by 234845** [comment]
I don't use gapps so I cannot say, but I guess you'd just use a LineageOS ROM with gapps already installed and follow the same procedure as a ROM without gapps.

**Comment by vog** [comment]
This is indeed a bit tricky, but let's discuss that in #5.

**Comment by on-fire** [comment]
Hello and thanks for your advice in advance.

I have downloaded H3113 firmware reduced.

If I flash reduced firmware with newflasher on my H3113 (it's running lineage 16.0 but ota-update points to "bluetooth keeps stopping") - does that fix "bluetooth keeps stopping" or do I have to perform an new installation of lineage with ait?

And should I exclude some files from flashing to use the existing installation of Lineage an its data further on?

**Comment by luk1337** [comment]
@on-fire just flash a zip file linked on https://forum.xda-developers.com/showpost.php?p=78973496&postcount=2.

**Comment by on-fire** [comment]
@luk1337 Thank you very much! This was, what I needed! I am following this issue-thread since beginning, but this solution posted from you above I didn't recognize...

ota-update is working fine now :)

**Comment by superbonaci** [comment]
I have same bluetooth issue on LG G2 D802, but can't open a new issue, please help.
